File: workflows/example_sync.py
from src.connectors.bitrix import BitrixConnector
from src.connectors.azure_custom import AzureCustomConnector
from src.connectors.llm.chain import LLMChain
from src.connectors.llm.providers.gemini import GeminiConnector
from src.connectors.llm.providers.openai import OpenAIConnector
from src.core.logger import log_step

async def main(payload: dict = None):
    """Ejemplo de flujo: Recibe de Bitrix -> Procesa IA -> Guarda Azure -> Responde Bitrix."""
    
    log_step("1. Inicializando Conectores")
    bitrix = BitrixConnector()
    azure = AzureCustomConnector()
    
    # Configuramos IA con Fallback
    ia = LLMChain(providers=[GeminiConnector(), OpenAIConnector()])
    
    log_step("2. Procesando datos con IA")
    content = payload.get("text", "Sin contenido") if payload else "Hola TAG"
    ai_response = await ia.ask(f"Analiza este requerimiento para TAG: {content}")
    log_step(f"AI Respondió: {ai_response[:100]}...")
    
    log_step("3. Guardando en Azure SharePoint")
    from src.core.config import settings
    await azure.upload_file(
        drive_id=settings.SHAREPOINT_DRIVE_ID,
        folder_path="Bitrix/IA",
        filename="ia_analysis.txt",
        content=ai_response.encode("utf-8")
    )
    
    log_step("4. Notificando a Bitrix")
    await bitrix.execute("crm.timeline.item.add", {"fields": {"COMMENT": f"IA dice: {ai_response}"}})
    
    return {"status": "completed", "ai": ai_response}

File: functions/src/workflows/example_sync.ts
import { BitrixConnector } from '../connectors/bitrix';
import { AzureCustomConnector } from '../connectors/azure_custom';
import { LLMChain } from '../connectors/llm/chain';
import { GeminiConnector } from '../connectors/llm/gemini';
import { OpenAIConnector } from '../connectors/llm/openai';
import { logStep } from '../core/logger';
import { defineString } from 'firebase-functions/params';

const sharepointDriveId = defineString('SHAREPOINT_DRIVE_ID');

export const main = async (payload: any = null) => {
  logStep("1. Inicializando Conectores (Node.js)");
  const bitrix = new BitrixConnector();
  const azure = new AzureCustomConnector();
  
  // Configuramos IA con Fallback
  const ia = new LLMChain([new GeminiConnector(), new OpenAIConnector()]);
  
  logStep("2. Procesando datos con IA");
  const content = payload?.text || "Hola TAG";
  const aiResponse = await ia.ask(`Analiza este requerimiento para TAG: ${content}`);
  logStep(`AI Respondió: ${aiResponse.substring(0, 100)}...`);
  
  logStep("3. Guardando en Azure SharePoint");
  await azure.uploadFile(
    sharepointDriveId.value(),
    "Bitrix/IA",
    "ia_analysis.txt",
    Buffer.from(aiResponse, "utf-8")
  );
  
  logStep("4. Notificando a Bitrix");
  await bitrix.execute("crm.timeline.item.add", { 
    fields: { COMMENT: `IA dice: ${aiResponse}` } 
  });
  
  return { status: "completed", ai: aiResponse };
};
